sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -0,0 +1,967 @@
+ *              .withConfiguration(myHadoopConfiguration); 
+ * }
+ * </pre>
+ * <P>
+ * // Read data with configuration and key translation (Example scenario: Coder is not available for
+ * key type hence key translation is required.).
+ * 
+ * <pre>
+ * {@code 
+ * p.apply("read",
+ *     HadoopInputFormatIO.<MyKeyType, InputFormatKeyClass>read() 
+ *              .withConfiguration(myHadoopConfiguration) 
+ *              .withKeyTranslation(myOutputKeyType);
+ * }
+ * </pre>
+ * <P>
+ * // Read data with configuration and value translation (Example scenario: Coder is not available
+ * for value type hence value translation is required.).
+ * 
+ * <pre>
+ * {@code 
+ * p.apply("read",
+ *     HadoopInputFormatIO.<InputFormatKeyClass, MyValueType>read() 
+ *              .withConfiguration(myHadoopConfiguration) 
+ *              .withValueTranslation(myOutputValueType);
+ * }
+ * </pre>
+ * 
+ * <P>
+ * // Read data with configuration, value translation and key translation (Example scenario: Coders
+ * are not available for both key type and value type of InputFormat hence key and value translation
+ * is required.).
+ * 
+ * <pre>
+ * {@code 
+ * p.apply("read",
+ *     HadoopInputFormatIO.<MyKeyType, MyValueType>read() 
+ *              .withConfiguration(myHadoopConfiguration) 
+ *              .withKeyTranslation(myOutputKeyType)
+ *              .withValueTranslation(myOutputValueType);
+ * }
+ * </pre>
+ * 
+ * <h3>Read data from Cassandra using HadoopInputFormatIO transform</h3>
+ * <p>
+ * To read data from Cassandra, CqlInputFormat can be used which needs following properties to be
+ * set.
+ * <p>
+ * Create Cassandra Hadoop configuration as follows:
+ * 
+ * <pre>
+ * {@code 
+ * Configuration cassandraConf = new Configuration();
+ *   cassandraConf.set("cassandra.input.thrift.port" , "9160");
+ *   cassandraConf.set("cassandra.input.thrift.address" , CassandraHostIp);
+ *   cassandraConf.set("cassandra.input.partitioner.class" , "Murmur3Partitioner");
+ *   cassandraConf.set("cassandra.input.keyspace" , "myKeySpace");
+ *   cassandraConf.set("cassandra.input.columnfamily" , "myColumnFamily");
+ *   cassandraConf.setClass("key.class" ,{@link java.lang.Long.class} , Object.class);
+ *   cassandraConf.setClass("value.class" ,{@link com.datastax.driver.core.Row.class} , Object.class);
+ *   cassandraConf.setClass("mapreduce.job.inputformat.class" ,{@link org.apache.cassandra.hadoop.cql3.CqlInputFormat.class} , InputFormat.class); }
+ * </pre>
+ * <p>
+ * Call Read transform as follows:
+ * 
+ * <pre>
+ * {@code
+ * PCollection<KV<Long,String>> cassandraData =
+ *          p.apply("read",
+ *                  HadoopInputFormatIO.<Long, String>read() 
+ *                      .withConfiguration( cassandraConf ) 
+ *                      .withValueTranslation( cassandraOutputValueType );
+ *   }
+ * </pre>
+ * <p>
+ * As default coder is available for CqlInputFormat key class i.e. {@link java.lang.Long} , key
```

Comment:
I ended up re-writing this paragraph. I think this works better:
* The CqlInputFormat key class is {@link java.lang.Long}, which has a Beam Coder. The 
 * CqlInputFormat value class is {@link com.datastax.driver.core.Row}, which does not have a Beam Coder.
 * Rather than write a coder, we will provide our own translation method: